src/domains/chart/components/lib-charts/dygraph-chart.tsx

```diff
@@ -331,6 +316,44 @@ export const DygraphChart = ({
   // state.dygraph_last_touch_end in old dashboard
   const dygraphLastTouchEnd = useRef<undefined | number>()
 
+  const dispatch = useDispatch()
+  const resetGlobalPanAndZoom = useCallback(() => {
+    latestIsUserAction.current = false // prevent starting panAndZoom
+    if (dygraphInstance) {
+      // todo on toolbox reset click, do updateOptions({ dateWindow: null })
+      // (issue existed also before rewrite)
+      dygraphInstance.updateOptions({
+        // reset dateWindow to the current
+        // @ts-ignore external typings dont support null
+        dateWindow: null,
+      })
+    }
+    dispatch(resetGlobalPanAndZoomAction())
+  }, [dispatch, dygraphInstance])
+
+  // setGlobalChartUnderlay is using state from closure (chartData.after), so we need to have always
+  // the newest callback. Unfortunately we cannot use Dygraph.updateOptions() (library restriction)
+  // for interactionModel callbacks so we need to keep the callback in mutable ref
+  const propsRef = useRef({
+    chartData,
+    globalChartUnderlay,
+    hoveredX,
+    resetGlobalPanAndZoom,
+    setGlobalChartUnderlay,
+    viewAfter,
+    viewBefore,
+  })
+  useLayoutEffect(() => {
+    propsRef.current.chartData = chartData
+    propsRef.current.hoveredX = hoveredX
+    propsRef.current.globalChartUnderlay = globalChartUnderlay
+    propsRef.current.resetGlobalPanAndZoom = resetGlobalPanAndZoom
+    propsRef.current.setGlobalChartUnderlay = setGlobalChartUnderlay
+    propsRef.current.viewAfter = viewAfter
+    propsRef.current.viewBefore = viewBefore
+  }, [chartData, globalChartUnderlay, hoveredX, resetGlobalPanAndZoom, setGlobalChartUnderlay,
+    viewAfter, viewBefore])
+
   useLayoutEffect(() => {
     if (chartElement && chartElement.current && !dygraphInstance) {
       const dygraphOptionsStatic = getInitialDygraphOptions({
```

Comment:
May be we can use this here?
`const {viewAfter, viewBefore} = propsRef.current`

Reply:
I don't understand, here i'm assigning necessary props to the `ref` because they are needed in old jquery library callbacks. I think i'm using destructuring few times here. Correct me please if i'm not on the same page :)